Assets/HoloToolkit-Examples/Utilities/Scripts/PlaneTargetGroupPicker.cs

```diff
@@ -18,6 +18,10 @@ public class PlaneTargetGroupPicker : Singleton<PlaneTargetGroupPicker>
 
         private Coroutine displayForSecondsCoroutine;
 
+        protected override void InitializeInternal()
```

Comment:
What's with this added override method that got added to all these irrelevant classes to this PR?

Reply:
This is introducing an abstract initialization step in singleton to help auto sort out dependencies between singletons.  Say you have three singletons where singleton A depends on singleton B and uses it for something in it's initialization step.  Singleton B, similarly depends on singleton C.  If initialization is done in Awake, since unity doesn't guarantee an order in which Awake is called, this could lead to singleton A being initialized first, and throwing errors, similarly with singleton B and C.  If you instead follow the practice of initializing in this function, initialization is done on demand, allowing Awake to be called in any order and initialization to be auto sorted, as long as there are no cycles in your dependency graph.  It also allows for lazy initialization of singletons.

Reply:
Again this should be it's own Pull Request.

Reply:
Ok, I can see that, I'll try to pull it out to its own pull request, and we can resolve that first then finish this one.